### DISCUSSION TITLE
How to build i386/x86 bun binary?

### DISCUSSION THREAD

**Discussion by tomasky** [oven-sh/bun#796] [General]
How to build i386/x86 bun binary ?    whether or not it is  supported

**Comment by Jarred-Sumner** [comment]
32-bit builds of Bun aren't supported and won't be for awhile. 

Several perf things depend on 64-bit pointers:
- FFI would need to be partially rewritten
- `Ref` used in the parser would need to be rewritten
- Pointer tagging would need to be turned off

It would be non-trivial

**Reply by tomasky** [reply]
got it ,thanks
